src/storage/src/source/types.rs

```diff
@@ -102,6 +106,62 @@ pub struct SourceMessage {
     pub metadata: Row,
 }
 
+mod columnation {
+    use columnation::{Columnation, Region};
+    use mz_repr::Row;
+
+    use super::SourceMessage;
+
+    impl Columnation for SourceMessage {
+        type InnerRegion = SourceMessageRegion;
+    }
+
+    #[derive(Default)]
+    pub struct SourceMessageRegion {
+        inner: <Row as Columnation>::InnerRegion,
+    }
+
+    impl Region for SourceMessageRegion {
+        type Item = SourceMessage;
+
+        unsafe fn copy(&mut self, item: &Self::Item) -> Self::Item {
```

Comment:
> This crate is wildly unsafe

😨 
(from the columnation [docstring](https://github.com/frankmcsherry/columnation/blob/master/src/lib.rs) )

Reply:
heh, yeah.. fortunately we will soon migrate to `flatcontainer` which is safer